tests/test_communication_manager.py

```diff
@@ -0,0 +1,282 @@
+    (9) Communication Manager schedules training job
+    (10) Communication Manager receives DMLResult for training job from
+         Scheduler
+    (11) Communication Manager gives DMLResult to Optimizer
+    (12) Optimizer updates its weights to trained weights
+
+    NOTE: These next steps are not implemented yet! (They need the COMM PR.)
+    (13) Optimizer tells Communication Manager to schedule a communication job
+    (14) Communication Manager schedules communication job
+    (15) Communication Manager receives DMLResult for communication job from
+         Scheduler
+    (16) Communication Manager gives DMLResult to Optimizer
+    (17) Optimizer tells the Communication Manager to do nothing
+
+    NOTE: Next steps are specific to Averaging PR
+    (18) Communication Manager receives new weights from Blockchain Gateway
+    (19) Communication Manager gives new weights to Optimizer
+    (20) Optimizer tells Communication Manager to schedule an averaging job
+    (21) Communication Manager schedules averaging job
+    (22) Communication Manager receives DMLResult for averaging job from
+        Scheduler
+    (23) Communication Manager gives DMLResult to Optimizer
+    (24) Optimizer updates its weights to initialized model and increments listen_iterations
+    (25) Optimizer tells Communication Manager to do nothing
+    (26) Communication Manager receives new weights from Blockchain Gateway
+    (27) Communication Manager gives new weights to Optimizer
+    (28) Optimizer tells Communication Manager to schedule an averaging job
+    (29) Communication Manager schedules averaging job
+    (30) Communication Manager receives DMLResult for averaging job from
+        Scheduler
+    (31) Communication Manager gives DMLResult to Optimizer
+    (32) Optimizer updates its weights to initialized model and increments listen_iterations
+    (33) Optimizer tells Communication Manager to schedule a training job since it's heard enough
+
+    """
+    communication_manager = CommunicationManager()
+    scheduler = DMLScheduler(config_manager)
+    communication_manager.configure(scheduler)
+    scheduler.configure(communication_manager)
+    true_job = make_initialize_job(make_model_json())
+    serialized_job = serialize_job(true_job)
+    new_session_event = {
+        "key": None,
+        "content": {
+            "optimizer_params": {},
+            "serialized_job": serialized_job
+        }
+    }
+    communication_manager.inform(
+        RawEventTypes.NEW_SESSION.name,
+        new_session_event
+    )
+    assert communication_manager.optimizer.job.job_type == JobTypes.JOB_INIT.name, \
+        "Should be ready to init!"
+    timeout = time.time() + 3
+    while time.time() < timeout and len(scheduler.processed) == 0:
+        # initialization job
+        scheduler.runners_run_next_jobs()
+        time.sleep(0.1)
+    assert len(scheduler.processed) == 1, "Initialization not completed in time!"
```

Comment:
_Carrying over previous thread since it was outdated, will resolve that thread_

You're right, but if the issue was originally Runner errors, then the outcome would still be a timeout (since failed jobs are discarded after three tries).

So then I think you should clarify in the messages that the issue could be an actual timeout or Runner error (since it's currently impossible to determine which is which?)